### PR TITLE
adding password, displayname, email in given step where users are created but not initialized

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

```    Given these users have been created but not initialized:``` doesn't add password, displayname and email information while creating user. This test make sure these informations are also added while creating user.

This test also includes test to make sure that a particular user is not displayed in the autocomplete list of sharing dialog.
## Related Issue

https://github.com/owncloud/phoenix/issues/1476

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4020/4

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4034/10

<p>





</p></details>

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 1]
```diff
@@ -7,22 +7,38 @@ const userSettings = require('../helpers/userSettings')
...
+    displayName = userSettings.getDisplayNameOfDefaultUser(userId)
+  }
+  if (!email) {
+    email = userSettings.getEmailAddressOfDefaultUser(userId)
+  }
+
+  body.append('userid', userId)
+  body.append('password', password)
+
+  console.log(displayName)
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: random debug output?

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 2]
```diff
@@ -7,22 +7,38 @@ const userSettings = require('../helpers/userSettings')
...
         const body = new URLSearchParams()
         body.append('key', 'display')
-        body.append('value', displayname)
-        const headers = httpHelper.createAuthHeader(client.globals.backend_admin_username)
+        body.append('value', displayName)
         fetch(`${client.globals.backend_url}/ocs/v2.php/cloud/users/${encodeURIComponent(userId)}?format=json`,
           { method: 'PUT', body: body, headers: headers }
         )
           .then(res => {
+            console.log(res.status)
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: random debug output?

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 3]
```diff
@@ -7,22 +7,38 @@ const userSettings = require('../helpers/userSettings')
...
 function createDefaultUser (userId) {
   const password = userSettings.getPasswordForUser(userId)
+  return createUser(userId, password)
+}
+
+function createUser (userId, password = false, displayName = false, email = false) {
+  const body = new URLSearchParams()
+
+  if (!displayName) {
+    displayName = userSettings.getDisplayNameOfDefaultUser(userId)
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: we might want to create a user without displayname, with that change its impossible to create a user that is in the default users table but should not have a display name

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 4]
```diff
@@ -120,8 +130,11 @@ Given('these users have been created but not initialized:', function (dataTable)
   return Promise.all(dataTable.rows().map((user) => {
     // TODO: handle displayname and email
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: todo done => delete the line

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4042/9

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4042/12

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4042/13

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4062/8

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4062/9

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4062/12

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4062/13

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4064/9

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4064/8

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4064/13

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4064/12

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4082/10

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4084/10

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4085/10

<p>





</p></details>

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 5]
```diff
@@ -7,55 +7,53 @@ const userSettings = require('../helpers/userSettings')
...
+          body.append('key', 'email')
+          body.append('value', email)
+          fetch(`${client.globals.backend_url}/ocs/v2.php/cloud/users/${encodeURIComponent(userId)}?format=json`,
+            { method: 'PUT', body: body, headers: headers }
+          )
+            .then(res => {
+              if (res.status !== 200) {
+                reject(new Error('Could not set email of user'))
+              }
+              resolve()
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: why no `resolve()` for displayname?

File: tests/acceptance/stepDefinitions/provisioningContext.js [hunk 6]
```diff
@@ -118,10 +121,12 @@ Given('these users have been created with default attributes but not initialized
 
 Given('these users have been created but not initialized:', function (dataTable) {
   return Promise.all(dataTable.rows().map((user) => {
-    // TODO: handle displayname and email
     let userId = user[0]
+    let password = user[1]
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/provisioningContext.js]: what is the order is different? can we use the headings here?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4118/9

<p>
















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4123/9

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4124/9

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4124/13

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4123/13

<p>















</p></details>

**Pull request merged**